gotradepal/backend/app/main.py

```
@@ -0,0 +1,73 @@
+    db.add(user); db.commit(); db.refresh(user)
+    return Token(access_token=create_access_token(str(user.id)))
+
+@app.post("/auth/login", response_model=Token)
+def login(email: str, password: str, db: Session = Depends(get_db)):
+    user = db.scalar(select(User).where(User.email == email))
+    if not user or not verify_password(password, user.hashed_password):
+        raise HTTPException(401, "Invalid credentials")
+    return Token(access_token=create_access_token(str(user.id)))
+
+@app.post("/profiles/{user_id}")
+def upsert_profile(user_id: str, payload: ProfileIn, db: Session = Depends(get_db)):
```

Ruff (0.15.21)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)